src/Maestro/Maestro.Web/Startup.cs

```diff
@@ -45,14 +49,89 @@ static Startup()
         {
             Triggers<BuildChannel>.Inserted += entry =>
             {
-                BuildChannel entity = entry.Entity;
                 DbContext context = entry.Context;
+                BuildChannel entity = entry.Entity;
+
+                StartAssociatedReleasePipelinesAsync(entry).GetAwaiter().GetResult();
+
                 var queue = context.GetService<BackgroundQueue>();
                 var dependencyUpdater = context.GetService<IDependencyUpdater>();
                 queue.Post(() => dependencyUpdater.StartUpdateDependenciesAsync(entity.BuildId, entity.ChannelId));
             };
         }
 
+        private static async Task StartAssociatedReleasePipelinesAsync(IInsertedEntry<BuildChannel, DbContext> entry)
```

Comment:
This method, even though called "Start" seems to actually wait. It's not named right. Its really "ExecuteAssociatedReleasePipelinesAsync".

I'm not sure we want a bunch of loops all running at once here, all waiting 15 seconds for it to finish.  One reboot of this service and we are screwed, we'll drop all of them on the floor.

This needs to be done reliably.  We need a store of "all releases that are pending" and we check them all and update the store and do whatever work is necessary.

Reply:
This logic needs to all be in another StatefulService so it can store information about release pipelines it has started and check when they are finished.